app/models/concerns/episode_media.rb

```diff
@@ -4,14 +4,93 @@ module EpisodeMedia
   extend ActiveSupport::Concern
 
   included do
+    enum :medium, [:audio, :uncut, :video, :override], prefix: true
+
     # NOTE: this just-in-time creates new media versions
     # TODO: convert to sql, so we don't have to load/check every episode?
     # TODO: stop loading non-latest media versions
     scope :feed_ready, -> { includes(media_versions: :media_resources).select { |e| e.feed_ready? } }
+
+    validate :validate_media_ready, if: :strict_validations
+
+    after_save :destroy_out_of_range_contents, if: ->(e) { e.segment_count_previously_changed? }
+    after_save :analyze_external_media
```

Comment:
But this isn't kicking off any analyze tasks right?  Maybe "create external media" would be more accurate?

Reply:
it does fire off an analyze task for the external media url, so we can get the duration, media type, and file size t include in the enclosure tag attributes

Reply:
It doesn't fire a task though, it just calls `create_external_media_resource`.  And later, `copy_media` fires the async job.  (That's the general pattern in Feeder - no async stuff fired unless you explicitly ask for it via `copy_media` or other means).